src/workerd/api/sockets.c++

```diff
@@ -254,7 +254,7 @@ jsg::Promise<void> Socket::close(jsg::Lock& js) {
       resolveFulfiller(js, nullptr);
       return js.resolvedPromise();
     });
-  }, [this](jsg::Lock& js, jsg::Value err) { return errorHandler(js, kj::mv(err)); });
+  }).catch_([this](jsg::Lock& js, jsg::Value err) { return errorHandler(js, kj::mv(err)); });
```

Comment:
```suggestion
  }).catch_(js, [this](jsg::Lock& js, jsg::Value err) { return errorHandler(js, kj::mv(err)); });
```
??

I can't remember off the top of my head if the `js` arg is required for `catch_` here or not.

Reply:
I.... have no idea why I didn't actually try to compile this before pushing it up.

Aaaand now I have to deal with the static_assert in `jsg::Promise<T>::catch_` seemingly complaining that the `catch_` handler is returning a `jsg::Promise<void>` rather than a plain `void` :( This may take a minute unless there's a common pattern I'm missing.

Reply:
I'm kind of confused how that hasn't caused issues before now -- shouldn't that assertion also accept `kj::isSameType<Promise<T>, ReturnType<ErrorFunc, Value, passLock>>()` in addition to `kj::isSameType<T, ReturnType<ErrorFunc, Value, passLock>>()` and then flatten the promise similarly to how kj promises work?

Anyways, unless you have a better solution I'm just going to add an error func to the abortPromise since that makes the types work out more easily.

Reply:
Sorry again for pushing up the broken code in the first place :(